src/codeflare_sdk/cluster/cluster.py

```diff
@@ -243,16 +245,21 @@ def cluster_dashboard_uri(self) -> str:
         Returns a string containing the cluster's dashboard URI.
         """
         try:
-            with oc.project(self.config.namespace):
-                route = oc.invoke(
-                    "get", ["route", "-o", "jsonpath='{$.items[*].spec.host}'"]
-                )
-                route = route.out().split(" ")
-                route = [x for x in route if f"ray-dashboard-{self.config.name}" in x]
-                route = route[0].strip().strip("'")
-            return f"http://{route}"
-        except:
-            return "Dashboard route not available yet, have you run cluster.up()?"
+            config.load_kube_config()
+            api_instance = client.CustomObjectsApi()
+            routes = api_instance.list_namespaced_custom_object(
```

Comment:
Ideally, to support running the SDK in-cluster, the dashboard URI would resolves to the dashboard service endpoint, and would not require Ingress / Route.